libraries/botbuilder-runtime/src/coreBotAdapter.ts

```diff
@@ -1,26 +1,86 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { BotFrameworkAdapter, ConversationState, useBotState, UserState } from 'botbuilder';
-import { AuthenticationConfiguration } from 'botframework-connector';
+import { AuthenticationConfiguration, ClaimsIdentity, SkillValidation } from 'botframework-connector';
+
+import {
+    ActivityEx,
+    BotAdapter,
+    BotFrameworkAdapter,
+    ConversationState,
+    InputHints,
+    MessageFactory,
+    TurnContext,
+    UserState,
+    useBotState,
+} from 'botbuilder';
 
 export class CoreBotAdapter extends BotFrameworkAdapter {
-    constructor(authConfig: AuthenticationConfiguration, conversationState: ConversationState, userState: UserState) {
+    constructor(
+        authConfig: AuthenticationConfiguration,
+        private readonly conversationState: ConversationState,
+        userState: UserState
+    ) {
         super({ authConfig });
 
         // attach storage?
         useBotState(this, userState, conversationState);
 
         this.onTurnError = async (turnContext, err) => {
-            console.error(err);
-
-            await turnContext.sendActivity(err.message, err.message);
+            console.error('[onTurnError] unhandled error', err);
 
-            try {
-                await conversationState.delete(turnContext);
-            } catch (err) {
-                console.error('Exception caught on attempting to delete conversation state', err);
-            }
+            await this.sendErrorMessage(turnContext, err);
+            await this.sendEoCToParentIfSkill(turnContext, err);
+            await this.clearConversationState(turnContext);
         };
     }
+
+    private async sendErrorMessage(turnContext: TurnContext, error: Error): Promise<void> {
```

Comment:
Not important at all, but in the JS SDK we've historically used `context` in place of `turnContext` for parameter names. In v5 let's align across all the SDKs in parameter names while accounting for being idiomatic according to language...